graph/pull.go

```diff
@@ -88,6 +88,10 @@ func (s *TagStore) Pull(image string, tag string, imagePullConfig *ImagePullConf
 		logrus.Debug("image does not exist on v2 registry, falling back to v1")
 	}
 
+	if utils.DigestReference(tag) {
+		return fmt.Errorf("pulling with digest reference failed from v2 registry, not falling back to v1")
```

Comment:
nit: Seems odd to tell users we're not falling back to v1. I don't think users care and it's confusing.